### ISSUE TITLE
Unresolved indirect/wildcard import of an extension trait when under #[async_trait]

### ISSUE THREAD

**Issue by Silvea12** [intellij-rust/intellij-rust#8768]
## Environment

* **IntelliJ Rust plugin version:** 0.4.168.4555-221
* **Rust toolchain version:** 1.60.0 (7737e0b5c 2022-04-04) x86_64-unknown-linux-gnu
* **IDE name and version:** CLion 2022.1 (CL-221.5080.224)
* **Operating system:** Linux 5.15.32-1-manjaro
* **Macro expansion engine:** new
* **Name resolution engine:** new
* **Additional experimental features:** org.rust.cargo.features.settings.gutter, org.rust.cargo.evaluate.build.scripts, org.rust.macros.proc

## Problem description
When creating a trait that uses `#[async_trait]` and importing via wildcard, the IDE considers the use of the trait's method as an unresolved reference unless it's a direct import - no wildcards.

See below code for how to reproduce. `Cargo.toml` only requires `async-trait` is added to the project. Any version should do, but my test uses `0.1.53`, the latest at time of writing.

## Steps to reproduce
```rs
mod my_type {
    pub struct MyType;
}

mod my_trait {
    use crate::my_type::MyType;
    use async_trait::async_trait;

    #[async_trait]
    pub trait MyTrait {
        async fn thing();
    }

    #[async_trait]
    impl MyTrait for MyType {
        async fn thing() {}
    }

}

use my_type::MyType;
// This reports an error in-editor, but still compiles
use my_trait::*;
// This works
// use my_trait::MyTrait;


fn main() {
    // ::thing() is seen as an undefined reference. Curiously, go to definition still resolves perfectly.
    MyType::thing();
}
```

![image](https://user-images.githubusercontent.com/3120639/164023443-cbea3edb-3345-4d46-985b-0ccb11ec0ef1.png)

**Comment by Silvea12** [comment]
Ah, it seems I had also reported this in a comment here and forgotten: https://github.com/intellij-rust/intellij-rust/issues/7863#issuecomment-1079936269

Still, good to have it as its own issue, as it's a different bug technically.

**Comment by Silvea12** [comment]
I found a workaround that works for the project this came up in, but it's not solving the underlying issue:
In my `prelude.rs`, I have this:
```rs
// Explicit include required to work around https://github.com/intellij-rust/intellij-rust/issues/8768
pub use crate::interaction_ext::{InteractionExt, *};
```
which seems to work for me. Just adding the extension trait directly to the prelude, everything else resolves from there. Less than ideal, but it does the job.
